service/history/shard/controller_impl.go

```diff
@@ -221,6 +221,9 @@ func (c *ControllerImpl) getOrCreateShardContext(shardID int32) (*ContextImpl, e
 		if shard.isValid() {
 			return shard, nil
 		}
+
+		shard.Unload()
+		delete(c.historyShards, shardID)
```

Comment:
shall we do the same at line 202?

Reply:
is 202 is reached, then the logic should continue to create a new shard
i mean adding additional logic at line 202 is not really necessary, but i can add

Reply:
actually, line 202 cannot delete the shard due to read lock.